test/puppetlabs/puppetdb/testutils/jetty.clj

```diff
@@ -61,42 +62,55 @@
     (spit logback-file (log-config log-file "ERROR"))
     [log-file (assoc-in config [:global :logging-config] logback-file)]))
 
-(defn current-port
-  "Given a trapperkeeper server, return the port of the running jetty instance.
-  Note there can be more than one port (i.e. SSL + non-SSL connector). This only
-  returns the first one."
-  [server]
-  (-> @(tka/app-context server)
-      (get-in [:WebserverService :jetty9-servers :default :server])
-      .getConnectors
-      first
-      .getLocalPort))
+(defn open-port-num
+  "Returns a currently open port number"
+  []
+  (with-open [s (java.net.ServerSocket. 0)]
+    (.getLocalPort s)))
+
+(defn assoc-open-port
+  "Updates `config` to include a (currently open) port number"
+  [config]
+  (assoc-in config [:jetty :port] (open-port-num)))
 
 (def ^:dynamic *server*)
 
 (defn puppetdb-instance
   "Stands up a puppetdb instance with `config`, tears down once `f` returns.
-  `services` is a seq of additional services that should be started in addition
-  to the core PuppetDB services. Binds *server* and *base-url* to refer to
-  the instance."
+  `services` is a seq of additional services that should be started in
+  addition to the core PuppetDB services. Binds *server* and
+  *base-url* to refer to the instance. `attempts` tracks how many
+  times an attempts were made to start Jetty (defaults to 1)"
   ([f] (puppetdb-instance (create-config) f))
   ([config f] (puppetdb-instance config [] f))
   ([config services f]
-   (let [[log-file config] (-> config conf/adjust-tk-config assoc-logging-config)
+   (puppetdb-instance config services 1 f))
+  ([config services attempts f]
```

Comment:
Minor quibble - I might be tempted to adjust this so that attempts counts down rather than up, and defaults to 10.  That way the (publically visible arg) is useful.  And then I might also be tempted to just hide the implementation via an internal recur (since you'll still need to track the count for the log message).  Other than that, looks good to me.